src/createRecords.ts

```diff
@@ -0,0 +1,64 @@
+import AirtableSDK from "airtable"
+
+import { SchemaZodT } from "./types/schema"
+import { SelectQueryParamsT } from "./types/queryParams"
+
+export interface CreateRecords<T extends SchemaZodT> {
+	tableId: string
+	baseId: string
+	schema: T
+	query?: SelectQueryParamsT
+	records: Array<T>
+}
+
+export interface CreateRecordsWithAirtable<T extends SchemaZodT>
+	extends CreateRecords<T> {
+	airtable: AirtableSDK
+}
+export interface CreateRecordsWithApiKey<T extends SchemaZodT>
+	extends CreateRecords<T> {
+	apiKey: string
+}
+
+//TODO should we change this to an upsert operation?
+export function createRecords<T extends SchemaZodT>({
```

Comment:
@iainkirkpatrick @hikurangi whats your opinion on just having upsert rather then Create and Update functions?

@fractalcactus sorry I'll start pinging you in these questions as well, you have have an opinion on the above too or anywhere else for that matter :)

Reply:
I just default to using Promises rather than async / await - but consumers of the library can use async / await, they're interchangeable from that perspective.

Reply:
i.e. as you have done in `airtable.test.ts`